locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+
+## Lifecycle of `.pipe()`
+
+To achieve a better understanding of backpressure, here is a flow-chart on the
+lifecycle of a [`Readable`][] stream being [piped][] into a [`Writable`][]
+stream:
+
+```javascript
++===============+               +===============+
+|   Your Data   |               |  .pipe(dest)  |
++===============+               |===============|
+        |                       |  Setup event  |
+        |                       |  closures:    |
++-------v-----------+           |               |         +-------------------+
+|  Readable Stream  >-----------+ * .data(cb)   +--------->  Writable Stream  |
++-^-------^-------^-+           | * .drain(cb)  |         +-------------------+
+  ^       ^       ^             | * .unpipe(cb) |                   |
+  |       ^       |             | * .error(cb)  |                   |
+  |       |       |             | * .finish(cb) |                   |
+  ^       |       |             | * .end(cb)    |           +=======v=========+
+  ^       |       ^             | * .close(cb)  |           |  .write(chunk)  |
+  |       |       ^             +---------------+           +=======+=========+
+  |       ^       |                                                 |
+  |       ^       |                              +------------------v---------+
+  ^       |       +-> if (!chunk)                |    Is this chunk too big?  |
+  ^       |       |     emit .end();             |    Is the queue busy?      |
+  |       |       +-> else                       +-------+----------------+---+
+  |       ^       |     emit .write();                   |                |
+  |       ^       ^                                   +--v---+        +---v---+
+  |       |       ^-----------------------------------<  No  |        |  Yes  |
+  ^       |                                           +------+        +---v---+
+  ^       |                                                               |
+  |       ^               emit .pause();          +=================+     |
+  |       ^---------------^-----------------------+  return false;  <-----+---+
+  |                                               +=================+         |
+  |                                                                           |
+  ^            when queue is empty     +============+                         |
+  ^------------^-----------------------<  Buffering |                         |
+               |                       |============|                         |
+               +> emit .drain();       |  ^Buffer^  |                         |
+               +> emit .resume();      +------------+                         |
+                                       |  ^Buffer^  |                         |
+                                       +------------+   add chunk to queue    |
+                                       |            <---^---------------------<
+                                       +============+
+```
```

Comment:
hi! this is so so good, i love it!

in terms of the diagram, i find the loop a little weird to get around and understand, although i definitely see where you're coming from.

What about making it a bit more linear? I am not good with replicating the fancy thing you have going, but something like this on paper? 

![stream-backpressure](https://cloud.githubusercontent.com/assets/8107784/23552713/1fc6969e-0015-11e7-9beb-e9f0fd4c6ae3.jpg)